components/VM_Arm/configurations/vm.h

```diff
@@ -68,22 +68,28 @@
     attribute int num_extra_frame_caps; \
     attribute int extra_frame_map_address; \
     attribute { \
-        string linux_ram_base; \
-        string linux_ram_paddr_base; \
-        string linux_ram_size; \
-        string linux_ram_offset; \
+        string ram_base; \
+        string ram_paddr_base; \
+        string ram_size; \
+        string ram_offset; \
         string dtb_addr; \
         string initrd_max_size; \
         string initrd_addr; \
-    } linux_address_config; \
+        string kernel_entry_addr = "0"; \
+    } vm_address_config; \
     attribute { \
-        string linux_name = "linux"; \
+        string kernel_name = "linux"; \
         string dtb_name = "linux-dtb"; \
         string initrd_name = "linux-initrd"; \
-        string linux_bootcmdline = ""; \
-        string linux_stdout = ""; \
+        string kernel_bootcmdline = ""; \
+        string kernel_stdout = ""; \
         string dtb_base_name = ""; \
-    } linux_image_config; \
+        int provide_dtb = true; \
+        int generate_dtb = false; \
```

Comment:
It seems `generate_dtb` overwrites `provide_dtb`, which is not intuitive. To make the configuration more clear, we could merge these should into a singe `dtb_handling`. Then we might even drop `dtb_base_name`and use `dtb_name` here also.
- `VM_DTB_FIXED (0)`: provide DTB from `dtb_name`. If unset don't provide a DTB. This is the default.
- `VM_DTB_GENERATE (1)`: generate based on `dtb_name`. If unset the generated DTB is based on an empty DTB
- `VM_DTB_NONE (2)` 2: don't provide DTB. `dtb_name` is ignored
- ... other options reserved for future use?

Reply:
I agree that DTB handling should be updated. This MR just updates the functionality to allow VMs to define their own per-VM settings. I like your suggestion, and I think it should be opened as an issue and made into its own MR.

Reply:
Do you have a dependency on the two variables already and want to get them in? Otherwise, I'd argue, that instead of introducing these two here now, just introduce `int dtb_handling` with the 3 values I suggested above. Then we don't introduce one thing just to remove it again later.

Reply:
I don't have a dependency on it, my concern is delaying the PR. 

This PR basically updates the functionality that already exists, where your suggested changes expand the functionality to something beyond the scope of this PR. If I can find time, I can implement this method, but I'd like to make it a separate PR.

Reply:
I added a warning print in the case where both `provide_dtb` and `generate_dtb` are set that warns the user that the provided dtb will not be loaded.